### PR TITLE
Optional account call limits

### PR DESCRIPTION
**Pull request by EgleHelms**
Optional env is: VITE_APP_ENABLE_ACCOUNT_LIMITS_ALL

Enables new choosing units and getting accordingly the new fields:
image

image

### CODE REVIEW

**Comment by davehorton** [comment]
When "sessions" are selected from the dropdown, are you going to change the labels e.g. to then read "Max Sessions" instead of "Max minutes"?

**Comment by EgleHelms** [comment]
I will check once more to make sure it is correct.

**Comment by EgleHelms** [comment]
You are right, changing it back is not working, need to resolve the state.

**Review by davehorton — APPROVED** [review]

**Comment by davehorton** [comment]
@kitajchuk if you have time please weigh in here, but we need this merged to progress so if you don't have time I will be merging today

**Comment by EgleHelms** [comment]
@davehorton fixed the typo, should be good now. I think this still needs a functional test. I keep getting unauthorized when I test with the main branch, not sure why.

File: src/components/forms/local-limits.tsx [hunk 1]
```diff
@@ -43,9 +48,65 @@ export const LocalLimits = ({
...
+        options={LIMIT_UNITS}
+        onChange={(e) => setUnit(e.target.value)}
+      />
+      {LIMITS.filter((limit) =>
+        unit === "sessions"
+          ? !limit.category.includes("minutes")
+          : limit.category.includes("minutes")
+      ).map(({ category, label }) => {
+        return (
+          <React.Fragment key={category}>
```
Comment (kitajchuk) [inline src/components/forms/local-limits.tsx:67]: Okay, so looking at this locally this is no good—this is a ton of copy pasted code. We always want to refactor in order to keep things DRY and not just dupe like code around. Here's my suggestion based on what is different and what is the same between the two.

Assign `filteredLimits` to a var conditionally based on the env and use that for the map: 

```tsx
const filteredLimits = import.meta.env.VITE_APP_ENABLE_ACCOUNT_LIMITS_ALL
    ? LIMITS.filter((limit) =>
        unit === "sessions"
          ? !limit.category.includes("minutes")
          : limit.category.includes("minutes")
      )
    : LIMITS.filter(
        (limit) =>
          !limit.category.includes("license") &&
          !limit.category.includes("minutes")
      );
```

Return a single render tree and just wrap the selector in an env condition and use `filteredLimits` for the map:

```tsx
return (
    <>
      {import.meta.env.VITE_APP_ENABLE_ACCOUNT_LIMITS_ALL && (
        <>
          <label htmlFor="units">Units</label>
          <Selector
            id="units"
            name="units"
            value={unit}
            options={LIMIT_UNITS}
            onChange={(e) => setUnit(e.target.value)}
          />
        </>
      )}
      {filteredLimits.map(({ category, label }) => {
        return (
          <React.Fragment key={category}>
            <label htmlFor={category}>{label}</label>
            <input
              ref={(el: HTMLInputElement) => {
                if (inputRef && inputRef.current) {
                  inputRef.current[category] = el;
                }
              }}
              id={category}
              type="number"
              name={category}
              placeholder="Enter quantity (0=unlimited)"
              min="0"
              value={updateLimitValue(category)}
              onChange={(e) => {
                const limit = localLimits.find((l) => l.category === category);
                const value = e.target.value ? Number(e.target.value) : "";

                if (limit) {
                  setLocalLimits(
                    localLimits.map((l) =>
                      l.category === category ? { ...l, quantity: value } : l
                    )
                  );
                } else {
                  setLocalLimits([
                    ...localLimits,
                    { category, quantity: value },
                  ]);
                }
              }}
            />
          </React.Fragment>
        );
      })}
    </>
  );
```

**Review by kitajchuk — CHANGES_REQUESTED** [review]
Some changes I'd like to see here 👍

**Comment by EgleHelms** [comment]
> Some changes I'd like to see here +1

applied the changes.

**Review by davehorton — APPROVED** [review]

**Review by kitajchuk — APPROVED** [review]

**Comment by davehorton** [comment]
@EgleHelms @kitajchuk I have merged the jambonz-api-server branch to main so in your further testing pls test against the main branch of jambonz-api-server

**Comment by davehorton** [comment]
@EgleHelms @kitajchuk it looks like review comments have been addressed?  Is this branch ready to merge?

**Comment by kitajchuk** [comment]
> @EgleHelms @kitajchuk it looks like review comments have been addressed? Is this branch ready to merge?

Yes smash the button 👍

**Pull request merged**